respy/simulate.py

```diff
@@ -460,22 +460,37 @@ def _random_choice(choices, probabilities, decimals=5):
     Here is an example with non-zero probabilities.
 
     >>> n_samples = 100_000
-    >>> choices = np.array([0, 1, 2])
+    >>> n_choices = 3
     >>> p = np.array([0.15, 0.35, 0.5])
     >>> ps = np.tile(p, (n_samples, 1))
-    >>> choices = _random_choice(choices, ps)
+    >>> choices = _random_choice(n_choices, ps)
     >>> np.round(np.bincount(choices), decimals=-3) / n_samples
     array([0.15, 0.35, 0.5 ])
 
     Here is an example where one choice has probability zero.
 
+    >>> choices = np.arange(3)
     >>> p = np.array([0.4, 0, 0.6])
     >>> ps = np.tile(p, (n_samples, 1))
-    >>> choices = _random_choice(choices, ps)
+    >>> choices = _random_choice(3, ps)
     >>> np.round(np.bincount(choices), decimals=-3) / n_samples
     array([0.4, 0. , 0.6])
 
     """
+    if isinstance(choices, int):
```

Comment:
WHy should we have so many cases/choices at such a low level? WOuldnt it be better to make the container of choices uniform at some point?

Reply:
Although the function is low-level, it is actually a vectorized replacement for ``numpy.random.choice``. It is only used in two different places, but needs to support ints and dict keys as choices. tuples and lists are added because it does not change anything. I think one could remove the ``np.ndarray`` check and the conversion to scalars at the end and defer some steps to where it is applied, but it reduces code and mental effort in other places.